tests/test_components/test_project_view/test_handgrading/test_handgrading.ts

```diff
@@ -637,6 +650,109 @@ describe('Comment tests', () => {
             [applied_long_description_annotation_1]);
         expect(delete_stub.calledOnce).toBe(true);
     });
+
+    test('Add applied annotations beyond max deduction', async () => {
+        for (let i = 1; i <= 3; ++i) {
+            let applied_annotation = data_ut.make_applied_annotation(
+                result,
+                annotation_empty_long_description,
+                {filename: 'file1.txt', first_line: i, last_line: i}
+            );
+            ag_cli.AppliedAnnotation.notify_applied_annotation_created(applied_annotation);
+        }
+        wrapper.vm.$forceUpdate();
```

Comment:
I tend to avoid `$forceUpdate` to make sure that the page is re-rendered appropriately. Does it work with `expect(await wait_until(wrapper, w => !w.vm.saving)).toBe(true);` instead?

Reply:
I will test with that. Is there a reason that `$forceUpdate` is used in 
```
 test('Handgraders allowed to leave custom comments', async () => {
        wrapper.vm.d_handgrading_result!.handgrading_rubric.handgraders_can_leave_comments = true;
        data_ut.set_global_user_roles(data_ut.make_user_roles({is_handgrader: true}));
        wrapper.vm.$forceUpdate();
        await wrapper.vm.$nextTick();
        expect(wrapper.find('#new-comment').exists()).toBe(true);

        expect(wrapper.findAll('.comment').at(0).find('.delete').exists()).toBe(true);
        expect(wrapper.findAll('.comment').at(3).find('.delete').exists()).toBe(true);
        // Can always delete applied annotations
        expect(wrapper.findAll('.comment').at(1).find('.delete').exists()).toBe(true);
        expect(wrapper.findAll('.comment').at(2).find('.delete').exists()).toBe(true);
    });
  ```?